### PR TITLE
Maven surefire not running and failing test and warnings

### PR DESCRIPTION
**Pull request by pricem14pc**
# What and why?

Unit tests weren't running as surefire plugin wasn't configured correctly. Only the failsafe integration tests were running.

Screenshot 2024-10-12 at 11 26 39

Also we were seeing various warnings and one of the tests was failing when run so there are changes to resolve those.

# How to test?

Check the unit tests are running and passing in Github Actions and locally using `mvn clean install`. There were ~158 unit tests not running previously.

Screenshot 2024-10-12 at 11 27 25

# Jira

### CODE REVIEW

File: src/test/java/uk/gov/ons/ctp/response/casesvc/state/StateTransitionManagerUnitTest.java [hunk 1]
```diff
@@ -584,14 +584,14 @@ public void givenUnknownEligibilityWhenEqLaunchThenComplete() throws CTPExceptio
   @Test
   public void testCaseGroupTransitionFromCompleteToNotStarted() throws CTPException {
     // Given
-    CaseGroupStatus notStarted = CaseGroupStatus.NOTSTARTED;
+    CaseGroupStatus notStarted = CaseGroupStatus.COMPLETE;
```
Comment (pricem14pc) [inline src/test/java/uk/gov/ons/ctp/response/casesvc/state/StateTransitionManagerUnitTest.java:587]:
```suggestion
    CaseGroupStatus caseGroupStatus = CaseGroupStatus.COMPLETE;
```

File: src/test/java/uk/gov/ons/ctp/response/casesvc/state/StateTransitionManagerUnitTest.java [hunk 2]
```diff
@@ -584,14 +584,14 @@ public void givenUnknownEligibilityWhenEqLaunchThenComplete() throws CTPExceptio
...
 
     // When
     CaseGroupStatus destinationState =
         caseGroupStateMachine.transition(
-            notStarted, CategoryDTO.CategoryName.OFFLINE_RESPONSE_PROCESSED);
+            notStarted, CategoryDTO.CategoryName.COMPLETED_TO_NOTSTARTED);
```
Comment (pricem14pc) [inline src/test/java/uk/gov/ons/ctp/response/casesvc/state/StateTransitionManagerUnitTest.java:592]:
```suggestion
            caseGroupStatus, CategoryDTO.CategoryName.COMPLETED_TO_NOTSTARTED);
```

**Review by arroyoAle — APPROVED** [review]
Looks good and tested, existing tests now run and pass

**Review by matthew-robinson-ons — APPROVED** [review]
Tests are now running and the build is working with the surefire updates

**Pull request merged**